app/forms/openid_connect_token_form.rb

```diff
@@ -85,7 +85,7 @@ def validate_code_verifier
   def validate_client_assertion
     return unless identity.present?
 
-    service_provider = ServiceProvider.new(client_id)
+    service_provider = ServiceProvider.find_by_issuer!(client_id)
```

Comment:
yes good catch.

I've been going back and forth on this. The old behavior was always a little loosey-goosey for my tastes, but keeping the consistency seems good.

How about a new method on `ServiceProvider`:

```rb
def from_issuer(issuer)
  where(issuer: issuer).first || ServiceProvider.new(issuer: issuer)
end
```

does that get at the old behaviour? WDYT?